server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -173,11 +173,13 @@ public SnapshotsService(Settings settings, ClusterService clusterService, IndexN
     /**
      * Same as {@link #createSnapshot(CreateSnapshotRequest, ActionListener)} but invokes its callback on completion of
      * the snapshot.
+     * Note: This method is only used in clusters that contain a node older than {@link #NO_REPO_INITIALIZE_VERSION} to ensure a backwards
+     * compatible path for initializing the snapshot in the repository is executed.
      *
      * @param request snapshot request
      * @param listener snapshot completion listener
      */
-    public void executeSnapshot(final CreateSnapshotRequest request, final ActionListener<SnapshotInfo> listener) {
+    public void executeSnapshotLegacy(final CreateSnapshotRequest request, final ActionListener<SnapshotInfo> listener) {
```

Comment:
This and the other `*Legacy` method rename (and the way they're selectively called from the transport action) is the only real difference between this commit and the original one that went into `master`. I realize that it is a little dirty to just duplicate everything and keep the old path code unchanged but IMO this is the easiest way to bring in the remaining snapshot back-ports cleanly.